packages/next/src/server/app-render/work-unit-async-storage.external.ts

```diff
@@ -39,7 +46,7 @@ export type RequestStore = {
 
   // DEV-only
   usedDynamic?: boolean
-}
+} & PhasePartial
```

Comment:
personally not a fan of use of partials for unioning. In this case it makes it possible for prerender stores to be constructed with or turned into action phase which should be impossible. It also makes it harder to see the complete type details in the IDE

Reply:
In reviewing further I see why prerender store needs action as a valid phase. Kind of unfortunate since this is only true in the route handler case. But still I think I'd just recommend we repeat the valid phases per store type

Reply:
yeah originally i only had `action` on RequestStore but in practice it seems like it's hard to narrow these